Add process for module building in Stream 9

mmathesius opened the pull request.

mmathesius commented:
Created this as a draft PR until the module building is made available to the community.

Zlopez commented on modules/ROOT/pages/techinfo/buildsystem.adoc:
```diff
@@ -10,13 +10,50 @@ This page covers the CentOS Stream Build System.
+
+* If your module does not yet exist in the
+https://gitlab.com/redhat/centos-stream/modules[CentOS Stream modules namespace in GitLab],
+open a https://issues.redhat.com/secure/CreateIssue!default.jspa?issuetype=3&pid=12326233[CS Jira]
```
I thought that the default place for the issues should be bugzilla.

mmathesius replied:
> I thought that the default place for the issues should be bugzilla.

Sure. That makes sense.

@bstinsonmhk said he had some thoughts about about filters and/or keywords to make sure such a BZ is directed to the proper queue. I imagine product="Red Hat Enterprise Linux 9" and version="CentOS Stream" is the starting point. What else?

Zlopez reviewed (commented):
Just one comment

mmathesius committed:
Add process for module building in Stream 9

Signed-off-by: Merlin Mathesius <mmathesi@redhat.com>

mmathesius commented:
Per discussions, I have dropped the steps for requesting creation of the module repos and importing the content. Those are Red Hat internal steps that don't belong in the contributor's guide.

I also rebased this to the latest commit in the main branch.

mmathesius commented:
Reverting this PR to draft state until module building is available to the community.
